### PR TITLE
Only callback after the connection is released

### PR DESCRIPTION
**Pull request by raymondfeng**
Fixes https://github.com/strongloop/loopback-next/issues/3296

#### Related issues

- connect to <link_to_referenced_issue>

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: example/app.js [hunk 1]
```diff
@@ -61,5 +61,5 @@ ds.discoverAndBuildModels(
         // ds.disconnect(); // This will crash node-oracle as the connection is disconnected while other statements are still running
       });
     });
-  }
+  },
```
Comment (bajtos) [inline example/app.js]: Can you please revert all unrelated changes where you are just adding a trailing comma to function args?

**Comment by hacksparrow** [comment]
@raymondfeng what about the accompanying tests?

File: example/app.js (hunk 1 above)
Comment (dhmlau) [reply example/app.js]: Is it coming from the linter?

Comment (raymondfeng) [reply example/app.js]: I had to fix eslint errors to pass CI.
